### PR TITLE
style(workspace-plugin): add thenable and promise resolving lint rules to catch promises leaks that cause issues when invoking generators

### PR DESCRIPTION
**Pull request by Hotell**
## Previous Behavior

Recently intermittent test failures started to happen caused by leaking promises etc
- https://uifabric.visualstudio.com/fabricpublic/_build/results?buildId=340958&view=logs&j=e51d397b-0a7b-5b89-4b48-0052e021371d&t=88c41216-94ff-5ca8-d686-48ce16c1eefc
- https://uifabric.visualstudio.com/fabricpublic/_build/results?buildId=340958&view=logs&j=e51d397b-0a7b-5b89-4b48-0052e021371d&t=88c41216-94ff-5ca8-d686-48ce16c1eefc
- https://uifabric.visualstudio.com/fabricpublic/_build/results?buildId=341095&view=logs&j=e51d397b-0a7b-5b89-4b48-0052e021371d&t=88c41216-94ff-5ca8-d686-48ce16c1eefc

## New Behavior

- sets lint rules to workspace-plugin project to catch unhandled promises
- fixes lint rule errors
- image

## Related Issue(s)

- Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: tools/workspace-plugin/.eslintrc.json [hunk 1]
```diff
@@ -10,15 +10,21 @@
     {
       "files": "*.json",
       "parser": "jsonc-eslint-parser",
-      "rules": {}
+      "rules": {
+        "@typescript-eslint/naming-convention": "off"
```
Comment (Hotell) [inline tools/workspace-plugin/.eslintrc.json:14]: this needs type information which for .json is not needed, thus we need to turn it off manually

File: tools/workspace-plugin/package.json [hunk 2]
```diff
@@ -7,6 +7,6 @@
   "scripts": {
     "test": "nx run @fluentui/workspace-plugin:test",
     "type-check": "nx run @fluentui/workspace-plugin:type-check",
-    "lint": "nx run @fluentui/workspace-plugin:lint-old"
```
Comment (Hotell) [inline tools/workspace-plugin/package.json:10]: this was not properly executing eslint with our specific configuration, moved to RAW binary. will be redone once we migrate to nx on pipelines

File: tools/workspace-plugin/project.json [hunk 3]
```diff
@@ -36,18 +36,14 @@
       }
     },
     "lint": {
-      "executor": "@nx/linter:eslint",
+      "executor": "@nx/eslint:lint",
```
Comment (Hotell) [inline tools/workspace-plugin/project.json:39]: dont use deprecated package

File: tools/workspace-plugin/src/generators/migrate-converged-pkg/index.ts [hunk 4]
```diff
@@ -137,7 +137,7 @@ function runMigrationOnProject(tree: Tree, schema: AssertedSchema, _userLog: Use
   // setup storybook
   setupStorybook(tree, options);
 
-  setupCypress(tree, options);
+  await setupCypress(tree, options);
```
Comment (Hotell) [inline tools/workspace-plugin/src/generators/migrate-converged-pkg/index.ts:140]: new lint rules saved the day

File: tools/workspace-plugin/src/generators/move-packages/index.spec.ts [hunk 5]
```diff
@@ -194,7 +192,6 @@ describe('move-packages generator', () => {
       packageJson = readJson(tree, packageJsonPath);
 
       expect(packageJson.scripts['build:local']).toEqual(
-        // eslint-disable-next-line @fluentui/max-len
```
Comment (Hotell) [inline tools/workspace-plugin/src/generators/move-packages/index.spec.ts:197]: unnecessary inline disables

**Review by marcosmoura — APPROVED** [review]

**Pull request merged**